src/app/app.component.ts

```diff
@@ -187,10 +236,31 @@ export class AppComponent implements OnInit {
       name: 'Item ' + this.itemCount,
       start: moment().startOf('day').add(5, 'h').toDate(),
       end: moment().add(3, 'days').endOf('day').add(5, 'h').toDate(),
-      classes: ''
+      classes: '',
+      client: '',
+      resource: '',
+      venue: '',
+      supplier: ''
     });
   }
 
+  showSectionId(e) {
+    let newSections;
+    if (e === 1) {
+      newSections = this.allSections.filter(sect => sect.name.includes('Supplier'));
+    }
+    if (e === 0) {
+      newSections = this.allSections.filter(sect => sect.name.includes('Resource'));
+    }
+    if (e === 2) {
+      newSections = this.allSections.filter(sect => sect.name.includes('Client'));
+    }
+    if (e === 3) {
+      newSections = this.allSections.filter(sect => sect.name.includes('Venue'));
+    }
+    this.sections = newSections;
+  }
+
```

Comment:
these need to be user defined fields, not hard coded.
You should spend a little bit more time understanding how the base plugin works so that this extension is generic enoguh that anyone can use it.
It needs to also still work as it does now at the same time.
Developers need to have the options on how they want to use the plugin.